packages/utils/upgrade/README.md

```diff
@@ -1 +1,179 @@
-<h1 align="center">upgrade</h1>
+# Strapi Upgrade Tool
+
+## Description
+
+The Strapi Upgrade Tool is the CLI for facilitating upgrades between Strapi versions.
+
+When run, the command will "step" from your current version of Strapi to the next version that includes code tranforms for breaking changes.
+
+From now on, all breaking changes, and ideally also deprecations, must have an accompanying code transform written to be accepted.
+
+### Types of Transforms
+
+The upgrade tool provides two types of transforms:
+
+- `json`: for updating a project's json files, primarily intended for the `package.json`
+- `code`: codemods; for updating a project's [js, jsx, ts, tsx] files
```

Comment:
We don't support jsx and tsx yet though, but we might consider adding it later

Reply:
Is it hard to support this? Feels like you're only gonna be able to migrate half a project realistically.

Reply:
Ahh, I thought jscodeshift supported it and we'd just have to check those filenames. For now I'll remove this and if we find a way to support it we'll add it back.

@joshuaellis are you aware of any front-end breaking changes that will need codemods at this point?

Reply:
Oh, you beat me by 13 minutes with an answer. We can do some testing, I would definitely like for it to be available as well and I suspect (and hope) it won't be very hard.

Reply:
Apparently it's supported, we will add it before the release :slightly_smiling_face:

Reply:
https://www.codeshiftcommunity.com/docs/react/